File: app/src/main/java/ua/com/foxminded/locationtrackera/model/locations/network/LocationsNetwork.java
package ua.com.foxminded.locationtrackera.model.locations.network;

import androidx.core.util.Pair;

import java.util.List;

import io.reactivex.rxjava3.core.Single;
import ua.com.foxminded.locationtrackera.model.locations.UserLocation;
import ua.com.foxminded.locationtrackera.util.Result;

public interface LocationsNetwork {

    Single<Result<Void>> sendLocations(List<UserLocation> locationList);

    Single<Result<?>> retrieveLocations(double startDate, double endDate);
}

File: models/src/main/java/ua/com/foxminded/locationtrackera/models/locations/network/LocationsNetwork.kt
package ua.com.foxminded.locationtrackera.models.locations.network

import io.reactivex.rxjava3.core.Single
import ua.com.foxminded.locationtrackera.models.locations.UserLocation
import ua.com.foxminded.locationtrackera.models.util.Result

interface LocationsNetwork {

    fun sendLocations(locationList: List<UserLocation>): Single<Result<Void?>>

    fun retrieveLocations(fromTime: Long, toTime: Long): Single<Result<List<UserLocation>>>
}
